force-app/classes/src/Params.cls

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+@IsTest
+public class Params {
+  public List<ArgumentMatcher> listOfArgs { get; private set; }
+
+  private Params() {
+    this.listOfArgs = new List<ArgumentMatcher>();
+  }
+
+  public void add(final Object callArgument) {
+    if (callArgument instanceof ArgumentMatcher) {
+      this.listOfArgs.add((ArgumentMatcher) callArgument);
+    } else {
+      this.listOfArgs.add(Matcher.equals(callArgument));
+    }
+  }
+
+  public static Params of() {
+    return new Params();
+  }
+
+  public static Params of(final Object arg) {
+    final Params params = of();
+    params.add(arg);
+    return params;
+  }
+
+  public static Params of(final Object arg1, final Object arg2) {
+    final Params params = of(arg1);
+    params.add(arg2);
+    return params;
+  }
+
+  public static Params of(
+    final Object arg1,
+    final Object arg2,
+    final Object arg3
+  ) {
+    final Params params = of(arg1, arg2);
+    params.add(arg3);
+    return params;
+  }
+
+  public static Params of(
+    final Object arg1,
+    final Object arg2,
+    final Object arg3,
+    final Object arg4
+  ) {
+    final Params params = of(arg1, arg2, arg3);
+    params.add(arg4);
+    return params;
+  }
+
+  public static Params of(
+    final Object arg1,
+    final Object arg2,
+    final Object arg3,
+    final Object arg4,
+    final Object arg5
+  ) {
+    final Params params = of(arg1, arg2, arg3, arg4);
+    params.add(arg5);
+    return params;
+  }
+
+  public static Params of(final List<Object> listOfArgs) {
```

Comment:
question/fix: is Apex compiler capable of pointing this method instead of `of(final Object arg)`?  or should we differentiate it through naming, with something like `ofList(final List<Object> listOfArgs)`?

+ what if you have a method with only 1 param that is a `List<Object>` ?

Reply:
Totally agree, plus I think we already had this issue before.
I'll wrote unit test and change it based on the result

Reply:
I think the `ParamsTest.givenListOf10MixType_whenOfIsCalledWithList_listOfArgsContains10Elements` assert covers that, and we have 100% coverage.
That being said, we can use `ofList` to show the difference in the API and I think we should do it